qiskit/ignis/experiments/calibration/cal_base_analysis.py

```diff
@@ -157,36 +159,44 @@ def run(self, **kwargs) -> any:
             any: the output of the analysis,
         """
         xvals, yvals = _create_data_vector(data=self.data,
-                                           metadata=self.metadata,
-                                           parameter=self.parameter,
-                                           series=self.series)
+                                           metadata_list=self.metadata)
 
         # fit for each initial guess
         result = None
-        for initial_guess in self.initial_guess(xvals, yvals):
-            p_opt, p_cov = optimize.curve_fit(self.fit_function,
-                                              xdata=xvals,
-                                              ydata=yvals,
-                                              p0=initial_guess,
-                                              bounds=self.fit_boundary(xvals, yvals))
-
-            # calculate chi square
-            chi_sq = _calculate_chisq(xvals=xvals,
-                                      yvals=yvals,
-                                      fit_yvals=self.fit_function(xvals, *p_opt),
-                                      n_params=len(initial_guess))
-            # calculate standard deviation
-            stdev = np.sqrt(np.diag(p_cov))
-
-            if result is None or result.chisq > chi_sq:
-                result = types.FitResult(fitval=p_opt,
-                                         stdev=stdev,
-                                         chisq=chi_sq,
-                                         xvals=xvals,
-                                         yvals=yvals)
-
-        # keep the best result
-        self._result = result
+        self._result = {}
+        for series_key, yvals in yvals.items():
```

Comment:
I think it doesn't work for ramsey xy fitting. you need to pass all yvals to the fitter to construct `series_x+i series_y`.

Reply:
I think there are two approaches. 1) keep this and overwrite this in ramsey fitter. 2) think  about more general solution. Iet me think about this.